### ISSUE TITLE
fix compilation for new kernel 5.6

### ISSUE THREAD

**Issue by joksik** [brandon-bailey/rtl8822bu#26]
Hi, i use Arch linux after update to kernel 5.6.0 rtl8822bu module cant build. logs:

DKMS make.log for rtl8822bu-20180723 for kernel 5.6.0-pf1 (x86_64)
sob, 4 kwi 2020, 07:14:14 CEST
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/5.6.0-pf1/build M=/var/lib/dkms/rtl8822bu/20180723/build  modules
make[1]: Wejście do katalogu '/usr/lib/modules/5.6.0-pf1/build'
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_cmd.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_security.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_debug.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_io.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_ioctl_query.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_ioctl_set.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_ieee80211.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_mlme.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_mlme_ext.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_mi.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_wlan_util.o
/var/lib/dkms/rtl8822bu/20180723/build/core/rtw_mlme_ext.c: In function ‘mgt_dispatcher’:
/var/lib/dkms/rtl8822bu/20180723/build/core/rtw_mlme_ext.c:1828:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
 1828 |    ptable->func = &OnAuthClient;
      |    ~~~~~~~~~~~~~^~~~~~~~~~~~~~~
/var/lib/dkms/rtl8822bu/20180723/build/core/rtw_mlme_ext.c:1830:2: note: here
 1830 |  case WIFI_ASSOCREQ:
      |  ^~~~
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_vht.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_pwrctrl.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_rf.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_recv.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_sta_mgt.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_ap.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/mesh/rtw_mesh.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_xmit.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_p2p.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_rson.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_tdls.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_br_ext.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_iol.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_sreset.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_btcoex_wifionly.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_btcoex.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_beamforming.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_odm.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_rm.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/rtw_rm_fsm.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/core/efuse/rtw_efuse.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/osdep_service.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/os_intfs.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/usb_intf.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/usb_ops_linux.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_linux.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/xmit_linux.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/mlme_linux.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/recv_linux.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/rtw_cfgvendor.o
  CC [M]  /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/wifi_regd.o
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c: In function ‘rtw_get_systime_us’:
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:690:18: error: storage size of ‘ts’ isn’t known
  690 |  struct timespec ts;
      |                  ^~
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:692:3: error: implicit declaration of function ‘getboottime’; did you mean ‘getboottime64’? [-Werror=implicit-function-declaration]
  692 |   getboottime(&ts);
      |   ^~~~~~~~~~~
      |   getboottime64
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:702:1: warning: control reaches end of non-void function [-Wreturn-type]
  702 | }
      | ^
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c: In function ‘cfg80211_rtw_change_iface’:
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:2363:10: warning: this statement may fall through [-Wimplicit-fallthrough=]
 2363 |   is_p2p = _TRUE;
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:2365:2: note: here
 2365 |  case NL80211_IFTYPE_STATION:
      |  ^~~~
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:2387:10: warning: this statement may fall through [-Wimplicit-fallthrough=]
 2387 |   is_p2p = _TRUE;
/var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.c:2389:2: note: here
 2389 |  case NL80211_IFTYPE_AP:
      |  ^~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:268: /var/lib/dkms/rtl8822bu/20180723/build/os_dep/linux/ioctl_cfg80211.o] Błąd 1
make[2]: *** Oczekiwanie na niezakończone zadania....
make[1]: *** [Makefile:1683: /var/lib/dkms/rtl8822bu/20180723/build] Błąd 2
make[1]: Opuszczenie katalogu '/usr/lib/modules/5.6.0-pf1/build'
make: *** [Makefile:2001: modules] Błąd 2

**Comment by p0ryae** [comment]
Yep. Right now the only kernel this driver is working on for me is linux510.

I hope they wake up and add compatibility for the new kernels like 511 and 512.
